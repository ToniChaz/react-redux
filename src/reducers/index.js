import { combineReducers } from 'redux'
import beers from './beer'
import modal from './modal'

const rootReducer = combineReducers({
  beers,
  modal
})

export default rootReducer
