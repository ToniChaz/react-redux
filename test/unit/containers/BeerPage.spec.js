import {mapStateToProps, mapDispatchToProps} from '../../../src/containers/BeerPage'
import { initialState } from '../../../src/reducers/modal'
import modalActions from '../../../src/actions/modal'
import beerActions from '../../../src/actions/beer'

describe('Containers', () => {
  test('should map modal state to props', () => {
    const state = {
      modal: initialState()
    }

    expect(mapStateToProps(state)).toMatchSnapshot()
  })

  test('should map modal actions to props', () => {
    //const beerActionPropNames = Object.keys(beerActions)
    const actionsPropNames = Object.keys(modalActions)

    expect(mapDispatchToProps(actionsPropNames)).toMatchSnapshot()
  })
})
