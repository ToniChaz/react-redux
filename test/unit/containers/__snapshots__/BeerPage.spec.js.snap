exports[`Containers should map modal actions to props 1`] = `
Object {
  "addBeer": [Function],
  "addBeerSuccess": [Function],
  "beerError": [Function],
  "closeModal": [Function],
  "fetchBeer": [Function],
  "fetchBeerSuccess": [Function],
  "openModal": [Function],
  "removeBeer": [Function],
  "removeBeerSuccess": [Function],
}
`;

exports[`Containers should map modal state to props 1`] = `
Object {
  "body": "",
  "callback": null,
  "extraParams": Object {},
  "open": false,
  "title": "",
}
`;
